src/lib/sql/sql_translator.hpp

```diff
@@ -85,6 +85,15 @@ class SQLTranslator final : public Noncopyable {
                                                        const std::shared_ptr<AbstractLQPNode>& input_node);
 
   std::shared_ptr<AbstractLQPNode> _translate_join(const hsql::JoinDefinition& select);
+
+  auto _classify_join_conditions(const hsql::JoinDefinition& join,
```

Comment:
I find the naming of these three functions imprecise and docs are missing.
_classify_join_conditions: What is being classified?
_collect_join_conditions: Collect? This breaks up a conjunctive chain into its components.
_include_join_conditions: This applies a predicate to a subplan